### PR TITLE
[INT-2461] Testing Updates

### PR DESCRIPTION
**Pull request by zemberdotnet**
# Description

This PR updates testing in `graph-cisco-amp` to be up to date with current patterns and test a larger set of data.

These changes will make testing easier going into adding `cisco_amp_vulnerabilities`.

## Changes:
 - `*.test.ts` files have been moved from `__tests__` directories to the directory of the file they test.
 - testing for the `synchronize` step used `executeStepWithDependencies`
   - `toMatchStepMetadata` can't be used because this step produces a mapped relationship, but it's still thoroughly tested. 
 - New `README.md` added in the `test` directory. This comes from the `integration-template`.
 - Schemas added in `constants.ts`
 - `test/context.ts` and `test/index.ts` have been removed as they were not needed.

### CODE REVIEW

File: src/steps/synchronize/index.test.ts
Comment (mdaum) [inline src/steps/synchronize/index.test.ts:71]: Note that you can use `toTargetEntities` to ensure each specific user_endpoint is being mapped to.
example: https://github.com/JupiterOne/graph-veracode-v2/blob/main/src/steps/findings/index.test.ts#L63

**Review by mdaum — APPROVED** [review]
See my comment on more targeted testing for mapped relationships. I won't hold up PR over it, but do give example I linked a look and make changes here if you feel it adds value.

File: src/steps/synchronize/index.test.ts
Comment (zemberdotnet) [reply src/steps/synchronize/index.test.ts:71]: It looks like `toTargetEntities` doesn't work with `targetFilterKeys` that are Arrays. This may or may not be a bug, but since it's already the existing behavior, I'm going to merge this and come back to it when I've found out if it's wrong to use an Array in `targetFilterKeys` or if `toTargetEntities` is bugged.

**Pull request merged**
